### PR TITLE
Update development tools: travis ci to github actions, tox to nox, nose to pytest

### PR DESCRIPTION
**Pull request by pietermarsman**
**Pull request**

Change the tools that we are using:

- Travis CI -> Github Actions (No recurring free compute)
- Tox -> Nox (More flexible, using python files to configure everything)
- Nose -> pytest (Noxe is not actively developped anymore)

Fix https://github.com/pdfminer/pdfminer.six/issues/498

**How Has This Been Tested?**

The CI pipeline is running again :) 

**Checklist**

- [x] I have added tests that prove my fix is effective or that my feature 
  works
- [x] I have added docstrings to newly created methods and classes
- [x] I have optimized the code at least one time after creating the initial 
  version
- [x] I have updated the [README.md](../README.md) or verified that this
  is not necessary
- [x] I have updated the [readthedocs](../docs/source) documentation or
  verified that this is not necessary
- [x] I have added a concise human-readable description of the change to
  [CHANGELOG.md](../CHANGELOG.md)

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -0,0 +1,95 @@
+# https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions
+# https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+#
+# Needed GitHub Repository Secret:
+#   PYPI_PASSWORD - PyPI API token allowing write, release access to PyPI.
+#     Limit scope to just project.
+#     (See package.python.org example link above.)
+
+name: On push
```
Comment (jstockwin) [inline .github/workflows/actions.yml:9]: I think this name will be displayed in a lot of places, how about `Continuous Integration` or something?

```suggestion
name: Continuous Integration
```

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+      - uses: actions/checkout@v2
+      - name: Set up Python ${{ env.default-python }}
+        uses: actions/setup-python@v2
+        with:
+          python-version: ${{ env.default-python }}
+      - name: Upgrade pip, Install nox
+        run: |
+          python -m pip install --upgrade pip
+          python -m pip install nox
+      - name: Lint
```
Comment (jstockwin) [inline .github/workflows/actions.yml:48]:
```suggestion
      - name: Mypy
```

File: .github/workflows/actions.yml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
```
Comment (jstockwin) [inline .github/workflows/actions.yml:93]:
```suggestion
      - name: Build Docs
```

File: .github/workflows/actions.yml [hunk 4]
```diff
@@ -0,0 +1,95 @@
+# https://help.github.com/en/actions/language-and-framework-guides/using-python-with-github-actions
+# https://packaging.python.org/guides/publishing-package-distribution-releases-using-github-actions-ci-cd-workflows/
+#
+# Needed GitHub Repository Secret:
+#   PYPI_PASSWORD - PyPI API token allowing write, release access to PyPI.
+#     Limit scope to just project.
+#     (See package.python.org example link above.)
```
Comment (jstockwin) [inline .github/workflows/actions.yml:7]: I assume this is copy + paste from somewhere, but actually `PYPY_PASSWORD` is not used, and this GitHub action will never publish to PyPI.

**Review by jstockwin — DISMISSED** [review]
In general looks nice! Good to have things upgraded.

Just a few small suggested changes, mostly typos.

In addition, this PR now has conflicts.

It may also be worth protecting the `develop` branch and requiring all of these actions to pass before merge. Don't do this until right before you merge this PR, though, as it'll block all others.

**Comment by pietermarsman** [comment]
@jstockwin thanks!

**Pull request merged**